`@std/async/deadline` with `@std/testing/time` doesn't work

**Describe the bug**

The `Promise` returned from the `@std/async@1.0.0/deadline` (but from `@std/async@0.224.0/deadline`) won't be fulfilled by `FakeTime` from `@std/testing/time`.

**Steps to Reproduce**

```ts
import { delay } from "jsr:@std/async@1.0.0/delay";
import { assertEquals } from "jsr:@std/assert@1.0.0";
import { deadline as deadlineOld } from "jsr:@std/async@0.224.0/deadline";
import { deadline as deadlineNew } from "jsr:@std/async@1.0.0/deadline";
import { FakeTime } from "jsr:@std/testing/time";

Deno.test("deadline v0.224.0", async () => {
  using t = new FakeTime();
  const p = deadlineOld(delay(1000), 10);
  let state: "pending" | "rejected" | "fulfilled" = "pending";
  p.then(() => {
    state = "fulfilled";
  }).catch(() => {
    state = "rejected";
  });
  await t.tickAsync(10);
  await t.runMicrotasks();
  assertEquals(state, "rejected");
});

Deno.test("deadline v1.0.0", async () => {
  using t = new FakeTime();
  const p = deadlineNew(delay(1000), 10);
  let state: "pending" | "rejected" | "fulfilled" = "pending";
  p.then(() => {
    state = "fulfilled";
  }).catch(() => {
    state = "rejected";
  });
  await t.tickAsync(10);
  await t.runMicrotasks();
  assertEquals(state, "rejected");
});
```

And the result is

```
running 2 tests from ./deadline_test.ts
deadline v0.224.0 ... ok (4ms)
deadline v1.0.0 ... FAILED (6ms)

 ERRORS 

deadline v1.0.0 => ./deadline_test.ts:21:6
error: AssertionError: Values are not equal.


    [Diff] Actual / Expected


-   pending
+   rejected


  throw new AssertionError(message);
        ^
    at assertEquals (https://jsr.io/@std/assert/1.0.0/equals.ts:47:9)
    at file:///D:/work/vim/denops.vim/deadline_test.ts:32:3

 FAILURES 

deadline v1.0.0 => ./deadline_test.ts:21:6

FAILED | 1 passed | 1 failed (15ms)

error: Test failed
```

**Expected behavior**

The test pass.


**Environment**

<!-- please complete the following information -->

- OS: macOS Sonoma
- deno version: v1.45.2
- std version: v1.0.0

--- a/testing/_time.ts
+++ b/testing/_time.ts
@@ -7,5 +7,5 @@ export const _internals = {
   clearTimeout,
   setInterval,
   clearInterval,
-  "AbortSignal_timeout": AbortSignal.timeout,
+  AbortSignalTimeout: AbortSignal.timeout,
 };

--- a/testing/_time.ts
+++ b/testing/_time.ts
@@ -7,4 +7,5 @@ export const _internals = {
   clearTimeout,
   setInterval,
   clearInterval,
+  "AbortSignal_timeout": AbortSignal.timeout,
 };

--- a/testing/time.ts
+++ b/testing/time.ts
@@ -220,7 +220,7 @@ function restoreGlobals() {
   globalThis.clearTimeout = _internals.clearTimeout;
   globalThis.setInterval = _internals.setInterval;
   globalThis.clearInterval = _internals.clearInterval;
-  AbortSignal.timeout = _internals.AbortSignal_timeout;
+  AbortSignal.timeout = _internals.AbortSignalTimeout;
 }
 
 function* timerIdGen() {

--- a/testing/time.ts
+++ b/testing/time.ts
@@ -197,12 +197,21 @@ function setTimer(
   return id;
 }
 
+function fakeAbortSignalTimeout(delay: number): AbortSignal {
+  const aborter = new AbortController();
+  fakeSetTimeout(() => {
+    aborter.abort(new DOMException("Signal timed out.", "TimeoutError"));
+  }, delay);
+  return aborter.signal;
+}
+
 function overrideGlobals() {
   globalThis.Date = FakeDate;
   globalThis.setTimeout = fakeSetTimeout;
   globalThis.clearTimeout = fakeClearTimeout;
   globalThis.setInterval = fakeSetInterval;
   globalThis.clearInterval = fakeClearInterval;
+  AbortSignal.timeout = fakeAbortSignalTimeout;
 }
 
 function restoreGlobals() {
@@ -211,6 +220,7 @@ function restoreGlobals() {
   globalThis.clearTimeout = _internals.clearTimeout;
   globalThis.setInterval = _internals.setInterval;
   globalThis.clearInterval = _internals.clearInterval;
+  AbortSignal.timeout = _internals.AbortSignal_timeout;
 }
 
 function* timerIdGen() {

--- a/testing/time_test.ts
+++ b/testing/time_test.ts
@@ -724,15 +724,15 @@ Deno.test("time.start returns the started time of the fake time", () => {
 });
 
 Deno.test("FakeTime doesn't affect AbortSignal.timeout unchanged if uninitialized", () => {
-  assertStrictEquals(AbortSignal.timeout, _internals.AbortSignal_timeout);
+  assertStrictEquals(AbortSignal.timeout, _internals.AbortSignalTimeout);
 });
 
 Deno.test("FakeTime fakes AbortSignal.timeout", () => {
   {
     using _time = new FakeTime(9001);
-    assertNotEquals(AbortSignal.timeout, _internals.AbortSignal_timeout);
+    assertNotEquals(AbortSignal.timeout, _internals.AbortSignalTimeout);
   }
-  assertStrictEquals(AbortSignal.timeout, _internals.AbortSignal_timeout);
+  assertStrictEquals(AbortSignal.timeout, _internals.AbortSignalTimeout);
 });
 
 Deno.test("FakeTime controls AbortSignal.timeout", () => {

--- a/testing/time_test.ts
+++ b/testing/time_test.ts
@@ -14,7 +14,7 @@ import { FakeTime, TimeError } from "./time.ts";
 import { _internals } from "./_time.ts";
 import { assertSpyCall, spy, type SpyCall } from "./mock.ts";
 
-function fromNow(): () => number {
+function fromNow(): (..._args: unknown[]) => number {
   const start: number = Date.now();
   return () => Date.now() - start;
 }
@@ -721,4 +721,52 @@ Deno.test("time.start returns the started time of the fake time", () => {
   using time = new FakeTime(1000);
   time.now = 2000;
   assertEquals(time.start, 1000);
+});
+
+Deno.test("FakeTime doesn't affect AbortSignal.timeout unchanged if uninitialized", () => {
+  assertStrictEquals(AbortSignal.timeout, _internals.AbortSignal_timeout);
+});
+
+Deno.test("FakeTime fakes AbortSignal.timeout", () => {
+  {
+    using _time = new FakeTime(9001);
+    assertNotEquals(AbortSignal.timeout, _internals.AbortSignal_timeout);
+  }
+  assertStrictEquals(AbortSignal.timeout, _internals.AbortSignal_timeout);
+});
+
+Deno.test("FakeTime controls AbortSignal.timeout", () => {
+  using time: FakeTime = new FakeTime();
+  const cb = spy(fromNow());
+  const expected: SpyCall[] = [];
+
+  const signal = AbortSignal.timeout(1000);
+  signal.onabort = () => cb();
+  time.tick(250);
+  assertEquals(cb.calls, expected);
+  time.tick(250);
+  assertEquals(cb.calls, expected);
+  time.tick(500);
+  expected.push({ args: [], returned: 1000 });
+  assertEquals(cb.calls, expected);
+  time.tick(2500);
+  assertEquals(cb.calls, expected);
+
+  assertEquals(signal.aborted, true);
+  assertInstanceOf(signal.reason, DOMException);
+  assertEquals(signal.reason.name, "TimeoutError");
+  assertEquals(signal.reason.message, "Signal timed out.");
+
+  const signalA = AbortSignal.timeout(1000);
+  signalA.addEventListener("abort", () => cb("a"));
+  const signalB = AbortSignal.timeout(2000);
+  signalB.addEventListener("abort", () => cb("b"));
+  const signalC = AbortSignal.timeout(1500);
+  signalC.addEventListener("abort", () => cb("c"));
+  assertEquals(cb.calls, expected);
+  time.tick(2500);
+  expected.push({ args: ["a"], returned: 4500 });
+  expected.push({ args: ["c"], returned: 5000 });
+  expected.push({ args: ["b"], returned: 5500 });
+  assertEquals(cb.calls, expected);
 });

--- a/testing/time_test.ts
+++ b/testing/time_test.ts
@@ -13,6 +13,7 @@ import {
 import { FakeTime, TimeError } from "./time.ts";
 import { _internals } from "./_time.ts";
 import { assertSpyCall, spy, type SpyCall } from "./mock.ts";
+import { deadline, delay } from "@std/async";
 
 function fromNow(): (..._args: unknown[]) => number {
   const start: number = Date.now();
@@ -769,4 +770,19 @@ Deno.test("FakeTime controls AbortSignal.timeout", () => {
   expected.push({ args: ["c"], returned: 5000 });
   expected.push({ args: ["b"], returned: 5500 });
   assertEquals(cb.calls, expected);
+});
+
+// https://github.com/denoland/std/issues/5499
+Deno.test("FakeTime regression test for issue #5499", async () => {
+  using t = new FakeTime();
+  const p = deadline(delay(1_000), 10);
+  let state: "pending" | "rejected" | "fulfilled" = "pending";
+  p.then(() => {
+    state = "fulfilled";
+  }).catch(() => {
+    state = "rejected";
+  });
+  await t.tickAsync(10);
+  await t.runMicrotasks();
+  assertEquals(state, "rejected");
 });

